rmw_fastrtps_cpp/src/rmw_node.cpp

```diff
@@ -307,10 +330,12 @@ rmw_destroy_node(rmw_node_t * node)
     result_ret = RMW_RET_ERROR;
   }
 
-  delete impl;
-
   Domain::removeParticipant(participant);
 
+  delete impl->listener;
+  impl->listener = nullptr;
+  delete impl;
```

Comment:
would `impl = nullptr;` make sense here ?

Reply:
Since `impl` is a local variable I don't think it would make sense.